p2panda-core/src/extensions.rs

```diff
@@ -61,10 +61,19 @@ use crate::Header;
 
 /// Trait definition of a single header extension type.
 pub trait Extension<T>: Extensions {
-    /// Extract the value of an extension based on it's type.
+    /// Extract the raw extension value of an extension based on it's type.
     fn extract(&self) -> Option<T> {
         None
     }
+
+    /// Extract the extension value with the option to derive it from material contained in the
+    /// passed header.
```

Comment:
I can imagine it's fairly abstract for someone to look at all of that functionality. Maybe we can provide some examples (not in Rust code but in "use cases") in the docs?